src/Components/Header/Header.tsx

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useCallback, useContext, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { TodosContext, USERS_URL, USER_ID } from '../Store/Store';
+import { Todo } from '../../Types/Todo';
+import { client } from '../../utils/fetchClient';
+
+type Props = {};
+
+export const Header: React.FC<Props> = React.memo(() => {
+  const {
+    todos,
+    setTodos,
+    setErrorMessage,
+    setTempItem,
+    creating,
+    setCreating,
+    focus,
+    updateTodo,
+  } = useContext(TodosContext);
+
+  const hasToggle = todos.length > 0;
+  const activeTodos = todos.filter(todo => !todo.completed);
+  const hasActiveTodos = activeTodos.length > 0;
```

Comment:
you can memoize this values too:
```suggestion
  const { hasToggle, activeTodos, hasActiveTodos } = useMemo(() => {
    const hasToggle = todos.length > 0;
    const activeTodos = todos.filter(todo => !todo.completed);
    const hasActiveTodos = activeTodos.length > 0;
    
    return { hasToggle, activeTodos, hasActiveTodos }
  }, [todos, activeTodos])
```

Reply:
![image](https://github.com/mate-academy/react_todo-app-with-api/assets/133979191/21cefac7-9a54-4f43-8e04-2d3569dd66b1)
I was able only add one depend and disable eslit for constants hasToggle, activeTodos and hasActiveTodos
`  const { hasToggle, activeTodos, hasActiveTodos } = useMemo(() => {
    const hasToggle = todos.length > 0;
    const activeTodos = todos.filter(todo => !todo.completed);
    const hasActiveTodos = activeTodos.length > 0;

    return { hasToggle, activeTodos, hasActiveTodos };
  }, [todos]);`